lib/galaxy/datatypes/interval.py

```diff
@@ -1447,6 +1449,104 @@ def sniff(self, filename):
+
+    def sniff(self, filename):
+        # Check that the file is compressed with bgzip (not gzip), i.e. the
+        # compressed format is BGZF, as explained in
+        # http://samtools.github.io/hts-specs/SAMv1.pdf
+        with open(filename, 'rb') as fh:
+            fh.seek(-28, 2)
+            last28 = fh.read()
+            if binascii.hexlify(last28) != b'1f8b08040000000000ff0600424302001b0003000000000000000000':
+                return False
+        # Check that the file has the required column name line in the
+        # commented header
+        file_iterator = util.compression_utils.get_fileobj(filename)
+        columns = None
+        for line in file_iterator:
+            if len(line) == 0 or line[0] != '#':
+                break
+            line = line.rstrip('\n\r')
+            if line.startswith('#columns: '):
+                columns = line.lower().replace('chrom', 'chr').replace('chromosome', 'chr').replace('stop', 'end').split()[1:]
+                break
+        if columns is None:
+            return False
+        # Make sure that the required columns are listed
+        return (('chr' in columns or 'chr1' in columns)
+                & ('pos' in columns or 'pos1' in columns or 'start' in columns or 'start1' in columns)
+                & ('chr2' in columns)
+                & ('pos2' in columns or 'start2' in columns))
+
+    def sniff_prefix(self, file_prefix):
+        return self.sniff(file_prefix.filename)
+
+    def set_meta(self, dataset, **kwd):
+        """ Creates the index for the pairix file. """
+        # We need to know which columns correspond to the chromosome name,
+        # the start coordinate, and if a stop coordinate is present.
+        # In order to produce a pairx index, we need to look at the
+        # required header line "#columns:". The sniffer should already
+        # have ensured that this line is present.
+        Tabular.set_meta(self, dataset, skip=0)
+        file_iterator = util.compression_utils.get_fileobj(dataset.file_name)
+        sep_character = None
+        comment_lines = 0
+        for line in file_iterator:
+            log.info(line)
+            if sep_character is not None:
+                break
+            elif len(line) > 0 and line[0] != '#':
+                fields = line.split()
+                if len(fields) > 1:
+                    sep_character = line.split(fields[1])[0].split(fields[0])[-1]
+            else:
+                comment_lines += 1
+                line = line.rstrip('\n\r')
+                if line.startswith('#columns: '):
+                    line = line.lower().replace('chrom', 'chr').replace('chromosome', 'chr').replace('stop', 'end')
+                    columns = line.split()[1:]
+        dataset.metadata.delimiter = sep_character
+        dataset.metadata.comment_lines = comment_lines
+        dataset.metadata.column_names = columns
+        dataset.metadata.columns = len(columns)
+        for i, name in enumerate(columns):
```

Comment:
```suggestion
        for i, name in enumerate(columns, 1):
```
and then drop the `+ 1` below. 

From the specs (https://github.com/4dn-dcic/pairix/blob/master/pairs_format_specification.md#standard-format) I understand the first 7 columns are required, reserved and in a fixed order, so can we assert that ?